ksql-engine/src/test/java/io/confluent/ksql/function/InternalFunctionRegistryTest.java

```diff
@@ -128,4 +139,26 @@ public Merger getMerger() {
         });
     assertThat(functionRegistry.getAggregate("my_aggregate", Schema.INT32_SCHEMA), not(nullValue()));
   }
+
+  @Test
+  public void shouldAddFunctionWithSameNameButDifferentReturnTypes() {
+    functionRegistry.addFunction(func);
+    assertTrue(functionRegistry.addFunction(
+        new KsqlFunction(Schema.INT64_SCHEMA,
+            Collections.singletonList(Schema.INT64_SCHEMA), "func", Object.class)));
+  }
+
+  @Test
+  public void shouldAddFunctionWithSameNameClassButDifferentArguments() {
```

Comment:
Should we also add a test case where we get a funtion with the same name, but without an acceptable list of arguments?

Reply:
what isn't an acceptable list of arguments? The acceptable arguments are all of Schema.Type and the List is `List<Schema.Type>` By definition you can only pass in acceptable arguments. Also, there is already a test that shows you can't add a function with the same name and argument types twice